controllers/capabilities/tenant_controller.go

```diff
@@ -117,12 +129,6 @@ func (r *TenantReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 			}
 		}
 
-		// add or remove finalizer, tenant CR will be deleted if tenant in 3scale does not exists, if tenant is deleted or if it's already marked for deletion
-		err = controllerhelper.ReconcileFinalizers(tenantR, r.Client, tenantFinalizer)
```

Comment:
Hey Eguzki. 
I understand how this fixes the CR being stuck on the cluster, maybe I am missing something but would this not mean that:
- if a tenant CR is deleted the CR will get deleted regardless of the tenant being removed from 3scale or not?
- if so do we really want this behavior? Wouldn't it be too misleading for users?

Reply:
The current implementation will stuck the k8s resource in "deleting" state and you cannot tell if that will eventually change or it is permanent and the CR will exist for ever even if the user tried to delete it. 

A k8s API call may fail, they fail, but eventually will success. With a 3scale API call, we cannot tell. You may try to fetch a tenant that does not exist internally and you will get error for ever. So, I consider safer, less problematic, even though not ideal, I agree, make sure the CR is deleted and have only one change to delete from 3scale.